modules/apps/login/login-authentication-opensso-web/src/main/java/com/liferay/login/authentication/opensso/web/internal/servlet/taglib/OpenSSOBottomJSPDynamicInclude.java

```
@@ -0,0 +1,155 @@
+import com.liferay.portal.kernel.portlet.LiferayPortletURL;
+import com.liferay.portal.kernel.portlet.LiferayWindowState;
+import com.liferay.portal.kernel.portlet.PortletURLFactoryUtil;
+import com.liferay.portal.kernel.security.sso.OpenSSO;
+import com.liferay.portal.kernel.servlet.taglib.BaseJSPDynamicInclude;
+import com.liferay.portal.kernel.servlet.taglib.DynamicInclude;
+import com.liferay.portal.kernel.settings.CompanyServiceSettingsLocator;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.PortletKeys;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.security.sso.opensso.configuration.OpenSSOConfiguration;
+import com.liferay.portal.security.sso.opensso.constants.OpenSSOConstants;
+
+import java.io.IOException;
+
+import javax.portlet.PortletRequest;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Stian Sigvartsen
+ */
+@Component(immediate = true, service = DynamicInclude.class)
+public class OpenSSOBottomJSPDynamicInclude extends BaseJSPDynamicInclude {
+
+	@Override
+	public void include(
+			HttpServletRequest request, HttpServletResponse response,
+			String key)
+		throws IOException {
+
+		long companyId = _portal.getCompanyId(request);
+
+		try {
+			OpenSSOConfiguration openSSOConfiguration = getOpenSSOConfiguration(
+				companyId);
+
+			if (!openSSOConfiguration.enabled()) {
+				return;
+			}
+		}
+		catch (Exception e) {
+			_log.error(e, e);
+
+			return;
+		}
+
+		HttpServletRequest originalHttpServletRequest =
+			_portal.getOriginalServletRequest(request);
+
+		String error = (String)originalHttpServletRequest.getAttribute(
+			"open.sso.error");
+
+		if (Validator.isBlank(error)) {
```

This should be the first check to fail fast. Getting configuration and checking if it's enabled is expensive.

---

Good idea. Thanks!